lib/ansible/module_utils/crypto.py

```diff
@@ -46,3 +48,68 @@ def get_fingerprint(path):
         pass
 
     return fingerprint
+
+
+def load_privatekey(path, passphrase=None):
+    """Load the specified OpenSSL private key."""
+
+    try:
+        privatekey_content = open(path, 'r').read()
+        privatekey = crypto.load_privatekey(crypto.FILETYPE_PEM,
+                                            privatekey_content,
+                                            passphrase)
+        return privatekey
+    except (IOError, OSError) as exc:
+        raise OpenSSLModuleError(exc)
+
+
+def load_certificate(path):
+    """Load the specified certificate."""
+
+    try:
+        cert_content = open(path, 'r').read()
+        cert = crypto.load_certificate(crypto.FILETYPE_PEM, cert_content)
+        return cert
+    except (IOError, OSError) as exc:
+        raise OpenSSLModuleError(exc)
+
+
+class OpenSSLModuleError(Exception):
+    pass
+
+
+class OpenSSLModule(object):
+
+    def __init__(self, module):
+        self.path = module.params['path']
+        self.state = module.params['state']
+        self.force = module.params['force']
+        self.name = os.path.basename(module.params['path'])
+        self.changed = True
+        self.check_mode = module.check_mode
+
+    def check(self):
+        """Ensure the resource is in its desired state."""
+
+        pass
+
+    def dump(self):
+        """Serialize the object into a dictionary."""
+
+        pass
+
+    def generate(self):
+        """Generate the resource."""
+
+        pass
+
+    def remove(self):
+        """Remove the resource from the filesystem."""
+
+        try:
+            os.remove(self.path)
+        except OSError as exc:
+            if exc.errno != errno.ENOENT:
+                raise OpenSSLModuleError(exc)
+            else:
+                self.changed = False
```

Comment:
Tying back in with what I wrote in the ```__init__``` about self.changed, setting self.changed = False here is dangerous.  A module might have already run a different method that made a change in state and set self.changed = True.  Then the code path in the main module might have accidentally run remove() twice.  The second time, self.changed will be set to False, leading to a False report that nothing was changed.

Setting self.changed = True on successful removal is a better way to do this.

Note, however, that modules will have to be careful.  For instance, they may both create and remove() a resource in one module run.  In that case, the module probably does not want to return changed due to the resource being removed().